fe/src/components/UI/molecules/header/OPHeader.tsx

```diff
@@ -36,6 +41,19 @@ const HEADER_VALUES: IHeaderValues[] = [
 ];
 
 const OPHeader = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const isLoggedIn = useSelector((store: RootState) => store.auth.loggedIn);
+
+  const logoutHandler = () => {
+    dispatch(logout());
+  };
+
+  useEffect(() => {
+    if (!isLoggedIn) navigate("/login");
+  }, [isLoggedIn]);
```

Comment:
This should be handled in navigation, not in each screen. But we will do this in another task later.